### DISCUSSION TITLE
Query and map one on relation table

### DISCUSSION THREAD

**Discussion by ckypper** [supabase/supabase#6517] [General]
How can i query and take only one variant record as an object not array with one element on relation table field

![image](https://user-images.githubusercontent.com/33152373/164385010-8e801c24-c667-4c75-a532-a01664eb2f1e.png)

**Comment by GaryAustin1** [comment]
Not quite sure what you are asking.  .single() will return an object if there is exactly 1 result of your query, otherwise error.  If you want to make sure you have just 1 record you could use .limit(1) in addition to .single().  If you can have 0 results then maybeSingle().

**Reply by ckypper** [reply]
single() is for items entity, the things i want is only one variant is return and  display as object, not array with 1 element.

**Reply by GaryAustin1** [reply]
Ah...  Even if you do table(count) it comes back as an array...
![image](https://user-images.githubusercontent.com/54564956/164873633-d42d0b77-50cb-4fc2-a5c2-5f38ba1a8016.png)
So I suspect not possible, but not a PostgREST expert.  If there is a positive answer it would be in postgrest.org or the github for it.
You may need to do an rpc function or just post process your result in javascript (what I would do).

**Reply by steve-chavez** [reply]
@ckypper Is the relationship between `items` and `variant` a [one-to-one relationship](https://pgolub.wordpress.com/2010/10/13/1-to-1-relationship-in-postgresql/)?

**Reply by ckypper** [reply]
> @ckypper Is the relationship between `items` and `variant` a [one-to-one relationship](https://pgolub.wordpress.com/2010/10/13/1-to-1-relationship-in-postgresql/)?

So i need define variant_id column in items table to make it be one-to-one right?

**Reply by steve-chavez** [reply]
Yes, but note that this is a [pending enhancement](https://github.com/PostgREST/postgrest/issues/1984). If that would solve your use case LMK and I'll work on it.
